smart_card_connector_app/src/window-devices-displaying.js

```diff
@@ -83,24 +85,25 @@ function displayReaderList(readers) {
   for (let reader of readers) {
     GSC.Logging.checkWithLogger(logger, readersListElement !== null);
     goog.asserts.assert(readersListElement);
-
+    
     var indicatorClasses = 'reader-state-indicator reader-state-indicator-' +
-                           reader['status'];
+    reader['status'];
     if (reader['status'] == GSC.PcscLiteServer.ReaderStatus.SUCCESS &&
-        reader['isCardPresent']) {
+    reader['isCardPresent']) {
       indicatorClasses = 'reader-card-present-indicator';
     }
     var indicator = goog.dom.createDom('span', indicatorClasses);
-
+    
     var indicatorContainer = goog.dom.createDom(
-        'span', 'reader-indicator-container', indicator);
-
-    var text = makeReaderNameForDisplaying(reader['name']) +
-        (reader['error'] ? ' (Error ' + reader['error'] + ')' : '');
-
-    var element = goog.dom.createDom('li', undefined, indicatorContainer, text);
-    goog.dom.append(readersListElement, element);
-  }
+      'span', 'reader-indicator-container', indicator);
+      
+      var text = makeReaderNameForDisplaying(reader['name']) +
+      (reader['error'] ? ' (Error ' + reader['error'] + ')' : '');
+      
+      var element = goog.dom.createDom('li', undefined, indicatorContainer, text);
+      goog.dom.append(readersListElement, element);
+    }
+    updateAddDeviceButtonText(readers.length);
```

Comment:
As discussed offline, let's, indeed, move this new call out from displayReaderList(), since it's a bit confusing that displayReaderList() not only updates the list, but also modifies the button title.

For instance, you could add a new function "onReadersChanged()" that would be passed to readerTrackerSubscriber() instead of displayReaderList(). onReadersChanged() would call displayReaderList() and updateAddDeviceButtonText().

Reply:
Will do.